components/loading-box/src/index.js

```diff
@@ -0,0 +1,166 @@
+      ${hexRgb(backgroundColor).red},
+      ${hexRgb(backgroundColor).green},
+      ${hexRgb(backgroundColor).blue},
+      ${backgroundColorOpacity})`,
+  },
+  '.fade-enter': {
+    ' > div': {
+      backgroundColor: `rgba(
+        ${hexRgb(backgroundColor).red},
+        ${hexRgb(backgroundColor).green},
+        ${hexRgb(backgroundColor).blue},
+        0)`,
+      transitionDuration: `${timeIn}ms`,
+    },
+    ' svg': {
+      opacity: 0,
+      transitionDuration: `${timeIn}ms`,
+      transitionDelay: `${timeIn / 2}ms`,
+    },
+  },
+  '.fade-enter-active': {
+    ' > div': {
+      backgroundColor: `rgba(
+        ${hexRgb(backgroundColor).red},
+        ${hexRgb(backgroundColor).green},
+        ${hexRgb(backgroundColor).blue},
+        ${backgroundColorOpacity})`,
+      transitionDuration: `${timeIn}ms`,
+    },
+    ' svg': {
+      opacity: 1,
+      transitionDuration: `${timeIn}ms`,
+      transitionDelay: `${timeIn / 2}ms`,
+    },
+  },
+  '.fade-exit': {
+    ' > div': {
+      backgroundColor: `rgba(
+        ${hexRgb(backgroundColor).red},
+        ${hexRgb(backgroundColor).green},
+        ${hexRgb(backgroundColor).blue},
+        ${backgroundColorOpacity})`,
+      transitionDuration: `${timeOut}ms`,
+    },
+    ' svg': {
+      opacity: 1,
+      transitionDuration: `${timeOut}ms`,
+    },
+  },
+  '.fade-exit-active': {
+    ' > div': {
+      backgroundColor: `rgba(
+        ${hexRgb(backgroundColor).red},
+        ${hexRgb(backgroundColor).green},
+        ${hexRgb(backgroundColor).blue},
+        0)`,
+      transitionDuration: `${timeOut}ms`,
+    },
+    ' svg': {
+      opacity: 0,
+      transitionDuration: `${timeOut}ms`,
+    },
+  },
+}));
+
+const Overlay = glamorous.div({
+  position: 'absolute',
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+  overflow: 'hidden',
+  height: '100%',
+  width: '100%',
+});
+
+const LoadingBox = ({
+  children,
+  backgroundColor,
+  backgroundColorOpacity,
+  loading,
+  spinnerColor,
+  timeIn,
+  timeOut,
+  ...props
+}) => (
+  <Wrapper>
+    <CSSTransition {...props} timeout={timeOut} classNames="fade" in={loading} unmountOnExit>
+      <Innerwrap
+        backgroundColor={backgroundColor}
+        backgroundColorOpacity={backgroundColorOpacity}
+        loading={loading}
+        timeIn={timeIn}
+        timeOut={timeOut}
+      >
+        <Spinner fill={spinnerColor} width="100px" height="100px" />
+        <Overlay />
+      </Innerwrap>
+    </CSSTransition>
+    {children}
+  </Wrapper>
+);
+
+LoadingBox.defaultProps = {
+  spinnerColor: WHITE,
+  backgroundColor: '#808080',
+  backgroundColorOpacity: 0.5,
+  loading: false,
+  timeIn: 200,
+  timeOut: 200,
+};
+
+LoadingBox.propTypes = {
+  children: PropTypes.node.isRequired,
+  spinnerColor: PropTypes.string,
+  // hex colour 3 or 6 characters with or without hash
+  backgroundColor: PropTypes.string,
+  backgroundColorOpacity: PropTypes.number,
+  loading: PropTypes.bool,
+  // number in milliseconds
```

Comment:
this text goes in to the docs, may be worth repeating for both timeIn and timeOut, also extend to say:

// length of fade in animation in milliseconds
// length of fade out animation in milliseconds